### PR TITLE
Add tf resource

### PR DESCRIPTION
**Pull request by Benjile**
Adds tf script

### CODE REVIEW

File: pkgs/tf.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    description = "wrapper around terraform";
+    homepage = "https://github.com/Caascad/tf";
+    license = licenses.mit;
+    maintainers = with maintainers; [ "bgeneze" ];
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
+    ls
+    pwd
```
Comment (eonpatapon) [inline pkgs/tf.nix]:
```suggestion
```
Remove theses lines

File: pkgs/tf.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{stdenv, source, fetchurl}:
```
Comment (eonpatapon) [inline pkgs/tf.nix]:
```suggestion
{stdenv, source}:  
```
`fetchurl` is not used

File: default.nix [hunk 3]
```diff
@@ -40,5 +40,7 @@ in {
 
   fly = pkgs.callPackage ./pkgs/fly.nix
     { source = sources.concourse; };
-
+  
+  tf  = pkgs.callPackage ./pkgs/tf.nix
+    { source = sources.tf; };
```
Comment (eonpatapon) [inline default.nix]:
```suggestion
    { source = sources.tf; inherit terraform; };
```

You need to pass `terraform` of the toolbox, if not the standard `terraform` of nixpkgs will be used

Comment (eonpatapon) [reply default.nix]: For that to work, make the attribute set recursive on line 22: `in rec {`

File: pkgs/tf.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    homepage = "https://github.com/Caascad/tf";
+    license = licenses.mit;
+    maintainers = with maintainers; [ "bgeneze" ];
+  };
+  buildInputs = [ makeWrapper ];
+  installPhase = ''
+    mkdir -p $out/bin
+    cp tf $out/bin/
+    chmod +x $out/bin/tf
+    wrapProgram $out/bin/tf $wrapperfile --prefix PATH ":" ${terraform}/bin
```
Comment (eonpatapon) [inline pkgs/tf.nix]:
```suggestion
    wrapProgram $out/bin/tf --prefix PATH ":" ${terraform}/bin
```
This should be enough

**Comment by eonpatapon** [comment]
Can you rebase on master and merge all commits into one ?

**Comment by Benjile** [comment]
yes but i do not want to merge. I would like to merge pr on tf first and release a 1.0.2 version of the script

**Pull request merged**